ls1intum/Artemis
# `Development`: Refactor result string generation

### Checklist
#### General


#### Server
#### Client
#### Changes affecting Programming Exercises

### Motivation and Context


The resultString got pretty complicated and the code used String manipulation to modify existing strings. This lead to some bugs and the string needs space in the database. Therefore it should be replaced.

### Description

This PR replaces the result string and only generates it on the client side using the existing attributes of the result.
To display the string for programming exercises correctly, ints store the amount of tests and code issues in the result.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- all Exercise types (also with manual assessment)

1. Log in to Artemis
2. Participate in the exercises
3. Grade the exercises
4. Complain on an exercise
Check the result String after each step that it displays the current information correctly 

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| result.service.ts | 73.58% | 73.73% |

### Screenshots


No/minimal visual changes
